src/main/java/com/information/user/UserController.java

```diff
@@ -23,4 +28,12 @@ public List<User> findUsers() {
     public List<User> findById(@PathVariable("id") int id) {
         return userService.findById(id);
     }
+
+    @PostMapping("/users")
+    public ResponseEntity<UserResponse> insert(@RequestBody UserRequest userRequest, UriComponentsBuilder uriBuilder) {
+        User user = userService.insert(userRequest.getName(), userRequest.getBirthdate());
```

Comment:
Requestクラスに、Userクラスのインスタンスを生成してreturnする`convertToUser `メソッドを作成して、以下のように呼んであげるのも手です！
```suggestion
        User user = userService.insert(userRequest.convertToUser());
```

Reply:
RequestUser クラスに新規でインスタンスメソッドを追加しました。
10cd5d6fe4eaaef952823b648bee9d6eaf8613e3

500エラーがレスポンスされるのでエラーログ等併せて共有させていただきます。

# Postman 画面
![image](https://github.com/kino41/RaiseTech_last/assets/155221768/fae7b36a-a85d-4281-89d9-f029b356905d)

# curl コマンド
curl --location 'http://localhost:8080/users' \
--header 'Content-Type: application/json' \
--data '{
 "name": "hiyoshi",
 "birthdate": "1950/06/22"
}'

# エラーログ
![image](https://github.com/kino41/RaiseTech_last/assets/155221768/4b216ffc-7dab-47e8-8649-3622abde10d1)

マークアップ部分に NullPointerException ： Cannot invoke "java.lang.Integer.intValue()" because "this.id" is null とあるので引数に設定している null に問題があるかと思っています。

# デバック実行
![image](https://github.com/kino41/RaiseTech_last/assets/155221768/6a4f35a5-55a4-423e-ac24-f612eddbab07)

画像の step over 後に、InvocationTargetException というクラスに自動的に遷移しており、UserController クラスの `buildAndExpand()` の引数に設定している `user.getId()` 部分がよくないように思えるのですが、 注目すべき箇所はあってますでしょうか？
```
URI location = uriBuilder.path("/users/{id}").buildAndExpand(user.getId()).toUri();
```